shared/src/TwinCollection.cs

```diff
@@ -176,6 +176,7 @@ public Metadata GetMetadata()
         /// Gets the LastUpdated time for this property
         /// </summary>
         /// <returns>DateTime instance representing the LastUpdated time for this property</returns>
+        /// <exception cref="System.ArgumentNullException">Thrown when _metadata is null</exception>
```

Comment:
_metadata is a private field that customers won't see. I'd put something to the effect of "Thrown when the TwinCollection metadata is null. An example would be when the TwinCollection class is created with the default constructor."